libraries/ESP8266WiFi/src/ArduinoWiFiServer.h

```diff
@@ -0,0 +1,127 @@
+/*
+  ArduinoWiFiServer.h - Arduino compatible WiFiServer
+  implementation for ESP8266Wifi library.
+  Copyright (c) 2020 Juraj Andrassy
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef arduinowifiserver_h
+#define arduinowifiserver_h
+
+#include <ESP8266WiFi.h>
+
+#ifndef MAX_MONITORED_CLIENTS
+#define MAX_MONITORED_CLIENTS 5
+#endif
+
+template <class TServer, class TClient>
+class ArduinoComptibleWiFiServerTemplate : public Server {
```

Comment:
Would it be wise to derive from `TServer` instead of `Server` ?

Reply:
it will be much better. it removes the redirecting methods and getServer